spec/operators/groupBy-spec.ts

```diff
@@ -1118,83 +1123,85 @@ describe('Observable.prototype.groupBy', () => {
   });
 
-  it('should return inners that when subscribed late exhibit hot behavior', () => {
-    const values = {
-      a: '  foo',
-      b: ' FoO ',
-      c: 'baR  ',
-      d: 'foO ',
-      e: ' Baz   ',
-      f: '  qux ',
-      g: '   bar',
-      h: ' BAR  ',
-      i: 'FOO ',
-      j: 'baz  ',
-      k: ' bAZ ',
-      l: '    fOo    '
-    };
-    const e1 = hot('-1--2--^-a-b-c-d-e-f-g-h-i-j-k-l-|', values);
-    const e1subs =        '^                         !';
-    const expected =      '--v---w---x-y-----z-------|';
-    const subv =          '       ^                   ';
-    const v =             '--------(d|)'               ;
-    const subw =          '               ^           ';
-    const w =             '----------------(h|)'       ;
-    const subx =          '                     ^     ';
-    const x =             '----------------------(k|)' ;
-    const suby =          '                              ^';
-    const y =             '------------------------------|';
-    const subz =          '                                ^';
-    const z =             '--------------------------------|';
-
-    const expectedGroups = {
-      v: Rx.TestScheduler.parseMarbles(v, values),
-      w: Rx.TestScheduler.parseMarbles(w, values),
-      x: Rx.TestScheduler.parseMarbles(x, values),
-      y: Rx.TestScheduler.parseMarbles(y, values),
-      z: Rx.TestScheduler.parseMarbles(z, values)
-    };
-
-    const subscriptionFrames = {
-      foo: Rx.TestScheduler.parseMarblesAsSubscriptions(subv).subscribedFrame,
-      bar: Rx.TestScheduler.parseMarblesAsSubscriptions(subw).subscribedFrame,
-      baz: Rx.TestScheduler.parseMarblesAsSubscriptions(subx).subscribedFrame,
-      qux: Rx.TestScheduler.parseMarblesAsSubscriptions(suby).subscribedFrame,
-      foo2: Rx.TestScheduler.parseMarblesAsSubscriptions(subz).subscribedFrame
-    };
-    const hasSubscribed = {};
-
-    const source = e1
-      .groupBy(
-      (val: string) => val.toLowerCase().trim(),
-      (val: string) => val,
-      (group: any) => group.skip(2)
-    )
-      .map((group: any) => {
-        const arr = [];
-
-        const subscriptionFrame = hasSubscribed[group.key] ?
-          subscriptionFrames[group.key + '2'] :
-          subscriptionFrames[group.key];
-
-        rxTestScheduler.schedule(() => {
-          group
-            .materialize()
-            .map((notification: Rx.Notification<any>) => {
-              return { frame: rxTestScheduler.frame, notification: notification };
-            })
-            .subscribe((value: any) => {
-              arr.push(value);
-            });
-          hasSubscribed[group.key] = true;
-        }, subscriptionFrame - rxTestScheduler.frame);
-
-        return arr;
-      });
-
-    expectObservable(source).toBe(expected, expectedGroups);
-    expectSubscriptions(e1.subscriptions).toBe(e1subs);
-  });
+  // HACK: I found this test hard to grok, and it was broken after the Subject refactor.
```

Comment:
FYI: I dropped the ball on this comment. @staltz, we'll need to circle back here, because I don't remember if there was any offline discussion about this test. I think we should capture what the purpose of this test was, and try to rewrite it to be a little more clear. This was broken by changes to how grouped subjects emitted, I believe, which was a cascading change from the Subject refactor to try to match Rx4 semantics.